crates/subspace-farmer/src/piece_cache.rs

```diff
@@ -325,10 +326,12 @@ impl PieceCache {
             });
         }
 
-        let mut element = vec![0; Self::element_size() as usize];
+        let mut element = BytesMut::zeroed(Self::element_size() as usize);
         if self.read_piece_internal(offset, &mut element)?.is_some() {
-            let mut piece = Piece::default();
-            piece.copy_from_slice(&element[PieceIndex::SIZE..][..Piece::SIZE]);
+            let element = element.freeze();
```

Comment:
Why do we need to call `freeze` on `BytesMut` when we have a similar call on `to_shared`?

Reply:
There is no `BytesMut::slice_ref`, only `Bytes::slice_ref` that we use to construct new `Bytes` from subset of the original `Bytes`. Yes, this means that technically piece allocation is a bit larger than it would be otherwise, but on the other hand we avoid extra allocation here and piece instances are short-lived anyway, so it makes more sense to do it this way.